### PR TITLE
Cleanups for check_imports.swift

### PR DESCRIPTION
**Pull request by paulb777**
Add main func, add ErrorLog class, and eliminate two globals

### CODE REVIEW

File: scripts/check_imports.swift [hunk 1]
```diff
@@ -57,19 +57,24 @@ let skipImportPatterns = [
   "FBLPromise",
 ]
 
-var foundError = false
+private class ErrorLogger {
```
Comment (wilhuff) [inline scripts/check_imports.swift:60]: If you wanted you could also pass the error logger to `checkFile`.

The benefit would be that tests could straightforwardly verify that certain inputs resulted in errors and that certain inputs did not without needing to add a way to clear the global error log.

File: scripts/check_imports.swift [hunk 2]
```diff
@@ -121,59 +127,65 @@ func checkFile(_ file: String, isPublic: Bool) {
...
         }
+        checkFile(fullTransformPath, inRepo: repoURL, isPublic: file.range(of: "/Public/") != nil)
       }
-      checkFile(fullTransformPath, isPublic: file.range(of: "/Public/") != nil)
     }
   }
 }
 
-exit(foundError ? 1 : 0)
+main()
```
Comment (wilhuff) [inline scripts/check_imports.swift:190]: perhaps have main return the status code?

File: scripts/check_imports.swift (hunk 1 above)
Comment (paulb777) [reply scripts/check_imports.swift:59]: Done

File: scripts/check_imports.swift (hunk 2 above)
Comment (paulb777) [reply scripts/check_imports.swift]: Done

**Pull request merged**